arelle/plugin/validate/HMRC/__init__.py

```diff
@@ -345,6 +354,10 @@ def checkFacts(facts):
                 modelXbrl.error("JFCVC.3312",
                     _("Facts are MANDATORY: %(missingItems)s"),
                     modelObject=modelXbrl, missingItems=", ".join(sorted(_missingItems)))
+            if not atLeastOneFacts and val.txmyType == "charities":
+                modelXbrl.error("JFCVC.3312.char",
+                                _("At least one of the facts is MANDATORY: %(missingItems)s"),
+                                modelObject=modelXbrl, missingItems=", ".join(sorted(mustHaveOneItem["charities"])))
```

Comment:
Similar to other suggestion, we can avoid hardcoding `"charities"` here and just use `val.txmyType in mustHaveOneItem`